drivers/spi/spi_nxp_s32.c

```diff
@@ -486,12 +495,29 @@ static int spi_nxp_s32_init(const struct device *dev)
 {
 	const struct spi_nxp_s32_config *config = dev->config;
 	struct spi_nxp_s32_data *data = dev->data;
-
+	uint32_t clock_rate;
 	uint8_t scaler, prescaler;
 
 	uint32_t ctar = 0;
 	int ret = 0;
 
+	if (!device_is_ready(config->clock_dev)) {
+		LOG_ERR("Clock control device not ready");
+		return -ENODEV;
+	}
+
+	ret = clock_control_on(config->clock_dev, config->clock_subsys);
+	if (ret) {
+		LOG_ERR("Failed to enable clock");
+		return ret;
+	}
+
+	ret = clock_control_get_rate(config->clock_dev, config->clock_subsys, &clock_rate);
+	if (ret) {
+		LOG_ERR("Failed to get clock frequency");
+		return ret;
+	}
+
```

Comment:
I wonder if it makes sense here if the clock_control_get_rate call fails to turn off the clock.

Reply:
I haven't seen that in other drivers consuming clock control. The clock source can also be shared with other peripherals, so I'd say it may cause more problems to turn off the clock. Let the drivers that need the clock fail to init instead.